File: 프로그래머스/1/12982. 예산/예산.js
function solution(d, budget) {
    var answer = 0;
    const dSort=d.sort((a,b)=>a-b);
    let sum=0;
    for(let i=0; i<d.length; i++){
        sum += dSort[i];
        if(sum>budget){
            return answer;
        }
        answer++;
    }
    return answer;
}

File: 프로그래머스/1/12982. 예산/예산.java
import java.util.*;

class Solution {
    public int solution(int[] d, int budget) {
        
        Arrays.sort(d);
        int answer = 0;
        
        for (int a : d){
              if (budget>=a){
            budget -= a;
            answer++;
              } else break;
        }
        return answer;
    }
}

